### PR TITLE
Add `ID` keyword, synonym for `IDENTIFICATION` (GC extension)

### PR DESCRIPTION
**Pull request by nberth**

### CODE REVIEW

File: test/output-tests/run_ml.expected [hunk 1]
```diff
@@ -1,4 +1,14 @@
...
+run_ml.at-122-prog.cob:21.13:
+  18          01  0SpecialTAGName  PIC X(3) VALUE "abc".
+  19   
+  20          01  employee.
+  21 >            05 id            PIC 9(1)  value 1.
+----                ^
+  22              05 name          PIC X(10) value "Someone".
+  23              05 dept          PIC X(10) value "Marketing".
+>> Hint: Missing .
+
```
Comment (emilienlemaire) [inline test/output-tests/run_ml.expected:11]: Are those errors expected?

File: test/output-tests/run_ml.expected [hunk 2]
```diff
@@ -9,925 +19,99 @@ run_ml.at-122-prog.cob:21.14-21.16:
...
+>> Hint: Missing DIVISION
 
-run_ml.at-122-prog.cob:89.42-89.46:
+run_ml.at-122-prog.cob:89.53-89.55:
   86              .
   87   
   88              *> Another test with mixed attributes and values
   89 >            XML GENERATE out FROM EMPLOYEE TYPE OF ID IS ATTRIBUTE
-----                                             ^^^^
+----                                                        ^^
```
Comment (emilienlemaire) [inline test/output-tests/run_ml.expected:97]: This also seems like an unwanted new error

**Review by emilienlemaire — COMMENTED** [review]
It seems like a few new errors got in, if you deem it ok to ignore them I'll approve the PR

File: test/output-tests/run_ml.expected (hunk 1 above)
Comment (nberth) [reply test/output-tests/run_ml.expected:11]: Yes, the actual command in the autotest file passes `-fnot-reserved=id`. We currently don't parse these arguments.

File: test/output-tests/run_ml.expected (hunk 2 above)
Comment (nberth) [reply test/output-tests/run_ml.expected:97]: Same reason as above.

**Review by emilienlemaire — APPROVED** [review]

**Pull request merged**